src/v1ToV6.js

```diff
@@ -0,0 +1,42 @@
+import parse from './parse.js';
+import { unsafeStringify } from './stringify.js';
+
+/**
+ * Convert a v1 UUID to a v6 UUID
+ *
+ * @param {string|Uint8Array} uuid - The v1 UUID to convert to v6
+ * @returns {string|Uint8Array} The v6 UUID as the same type as the `uuid` arg
+ * (string or Uint8Array)
+ */
+export default function v1ToV6(uuid) {
+  const v1Bytes = typeof uuid === 'string' ? parse(uuid) : uuid;
+
+  const v6Bytes = _v1ToV6(v1Bytes);
+
+  return typeof uuid === 'string' ? unsafeStringify(v6Bytes) : v6Bytes;
+}
+
+// Do the field transformation needed for v1 -> v6
+function _v1ToV6(v1Bytes, randomize = false) {
+  return Uint8Array.of(
+    ((v1Bytes[6] & 0x0f) << 4) | ((v1Bytes[7] >> 4) & 0x0f),
+    ((v1Bytes[7] & 0x0f) << 4) | ((v1Bytes[4] & 0xf0) >> 4),
```

Comment:
Hi @broofa, i'm just curious, but in this case, is there any difference doing `>> 4` and then `& 0x0f`, compared to `& 0xf0` and then `>> 4`?

i noticed that you did the former on the first line, while the rest uses the latter.

Reply:
No, I believe both ways are functionally equivalent.  'Not sure why I wrote it this way.  It'd probably be good to switch it to `&0xf0) >> 4)` for consistency at some point, but not something I'm going to lose sleep over. :-)